query-engine/connectors/sql-query-connector/src/query_builder/write.rs

```diff
@@ -78,6 +82,8 @@ pub fn create_records_nonempty(
     let insert = values.into_iter().fold(insert, |stmt, values| stmt.values(values));
     let insert: Insert = insert.into();
 
+    let insert = insert.append_trace(&Span::current());
```

Comment:
I prefer these grouped when you're shadowing:
```suggestion
    let insert = values.into_iter().fold(insert, |stmt, values| stmt.values(values));
    let insert: Insert = insert.into();
    let insert = insert.append_trace(&Span::current());
```